src/iosrv/srv.c

```diff
@@ -770,23 +806,40 @@ dss_xstreams_init()
 	D_DEBUG(DB_TRACE, "%d cores detected, starting %d main xstreams\n",
 		dss_core_nr, dss_tgt_nr);
 
-	for (i = 0; i < DSS_XS_NR_TOTAL; i++) {
-		hwloc_obj_t	obj;
-
-		obj = hwloc_get_obj_by_depth(dss_topo, dss_core_depth,
-					     i % dss_core_nr);
-		if (obj == NULL) {
-			D_ERROR("Null core returned by hwloc\n");
-			D_GOTO(failed, rc = -DER_INVAL);
-		}
+	xstream_data.xd_xs_nr = DSS_XS_NR_TOTAL;
+	/* start system service XS */
+	for (i = 0; i < dss_sys_xs_nr; i++) {
+		xs_id = i;
+		rc = dss_start_xs_id(xs_id);
+		if (rc)
+			D_GOTO(out, rc);
+	}
 
-		rc = dss_start_one_xstream(obj->allowed_cpuset, i);
+	/* start main IO service XS */
+	for (i = 0; i < dss_tgt_nr; i++) {
+		xs_id = DSS_MAIN_XS_ID(i);
+		rc = dss_start_xs_id(xs_id);
 		if (rc)
-			D_GOTO(failed, rc);
+			D_GOTO(out, rc);
 	}
-	D_DEBUG(DB_TRACE, "%d execution streams successfully started\n",
-		dss_tgt_nr);
-failed:
+
+	/* start offload XS if any */
+	if (dss_tgt_offload_xs_nr == 0)
+		D_GOTO(out, rc);
+	for (i = 0; i < dss_tgt_nr; i++) {
+		int j;
+
+		for (j = 0; j < dss_tgt_offload_xs_nr; j++) {
+			xs_id = DSS_MAIN_XS_ID(i) + j + 1;
+			rc = dss_start_xs_id(xs_id);
```

Comment:
While we are there, it would be great to use pthread_setname_np() to set a proper name to each pthread. If possible, I would suggest to use the following naming scheme:
- the md thread could just be called daos_md
- the target thread could just be called daos_tgt${context_id}_${thread_id} where context_id is effectively the target number and thread_id starts with 1 for the primary tgt xstream and is then bumped for each offload xstream attached to this target.

Makes sense?

Reply:
fine to me, I'll add it. thx